### ISSUE TITLE
Auto Post not working 

### ISSUE THREAD

**Issue by PeterDredd** [10up/autoshare-for-twitter#315]
### Describe your question

Hello,

I recently integrated the auto post to x plugin and got the successful notification after finishing the setup.

But after publishing a post.. it doesn’t auto post to my X account. I checked the box that says post to Twitter.. but it still didn’t work.

![IMG_3299](https://github.com/10up/autoshare-for-twitter/assets/164010930/734c9356-18f6-45f6-99cd-0e6dfa8ae8e0)
![IMG_3305](https://github.com/10up/autoshare-for-twitter/assets/164010930/7003c530-2aef-409d-b305-e2b8e48d30e3)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by PeterDredd** [comment]
Please it’s urgent.. somebody share a tip on how to fix this.

![Uploading IMG_3411.jpeg…]()

**Comment by PeterDredd** [comment]
Someone should help out please..  I have a verified twitter account if that means anything. 

![IMG_3411](https://github.com/10up/autoshare-for-twitter/assets/164010930/ac1f8a9c-baaa-453c-a50d-e5e28b0b8ffe)

**Comment by PeterDredd** [comment]
@iamdharmesh please help look into this

**Comment by iamdharmesh** [comment]
Hi @PeterDredd,

Could you please ensure that the "Post to X/Twitter" checkbox is selected when you publish the post? Based on the screenshot you provided, it appears to be unchecked. Also, could you please manually attempt to post to X by clicking "Post to X/Twitter" and provide the details of any error you encounter?

Providing error details will assist us in identifying the root cause and resolving the issue.

Thank you.

**Comment by PeterDredd** [comment]
@iamdharmesh 

I have it checked.. that’s in the mobile view.. I’ve attached image from desktop view.

When I try to share manually by clicking ‘Post to X/Twitter’ the dot icon loads for a few seconds and stops… showing no error and the post stays unshared. 

I’ve disconnected and reconnected.. still didn’t work.
Images attached 

![Image 20-03-2024 at 18 06](https://github.com/10up/autoshare-for-twitter/assets/164010930/4baa55a8-76fe-4b25-b566-0eb58bc0370c)
![Image 20-03-2024 at 18 10](https://github.com/10up/autoshare-for-twitter/assets/164010930/336b1675-20dd-4fb0-9e31-0b4834ec6bfa)
![Image 20-03-2024 at 18 11](https://github.com/10up/autoshare-for-twitter/assets/164010930/7469ab0a-8578-4fb5-810c-8e12937f146a)

**Comment by iamdharmesh** [comment]
Thanks for the further details, @PeterDredd. Since there are no errors, it seems there are some conflicts with other plugins. Some next steps here could be:

1. Check log files to see if any useful errors are logged there. If you are not familiar with how to check logs, this article can help: https://www.wpbeginner.com/wp-tutorials/how-to-find-and-access-wordpress-error-logs-step-by-step/.
2. Deactivate other plugins temporarily to identify if the issue is due to conflicts between plugins.

Please let me know if following these steps provides some useful information, and we can make this work.

Thank you

**Comment by PeterDredd** [comment]
So I finally took the bold step to deactivate all plugins.. using file manager in cpanel.. I was reluctant because some of the plugins are serving real time purposes.

Deactivated all and only activated the share to Twitter plugin.. 

Went to a drafted article… published it.. scrolled down to check the status on auto post to twitter.. 

Same as it was before “this post post has not been posted”

Did the manual posting.. it rolled and stopped with no error message just as I complained before.

In all of this, it’s only auto post to Twitter plug-in that’s active.

In short.. deactivating all plugins didn’t work.. 
no useful error in log files..

Any other way out?

**Comment by PeterDredd** [comment]
I still need help with this

**Comment by PeterDredd** [comment]
@iamdharmesh @jamesmorrison @felipeelia @cadic 

How do I fix this?

**Comment by iamdharmesh** [comment]
@PeterDredd, apologies for the delayed response. Could you please try setting up the plugin in a fresh WP installation with the default theme (just for testing) and try to post? It will help identify whether the issue lies with the website or the account setup.

Thanks.

**Comment by PeterDredd** [comment]
So I tried another plugin.. looking to see if I’ll get a response error .. and yes I did. 

I got “API usage rate limit exceeded” meanwhile I haven’t used any this year.. not just this month.. 

The monthly cap usage is showing 0 posts pulled of 1,500

![Screenshot 2024-04-13 at 16 34 54](https://github.com/10up/autoshare-for-twitter/assets/164010930/790c4f6a-2b0a-48d8-bbe4-1a5b6ee3b526)
![IMG_6395](https://github.com/10up/autoshare-for-twitter/assets/164010930/c862d36a-9f53-4aca-b6f7-bcdd8141351c)

Please how do I fix this?

**Comment by iamdharmesh** [comment]
Hi @PeterDredd, Thank you for checking on this. I'm not sure which plugin you are using, but I believe the error depends on how the plugin utilizes the API. Because of this, the error on our plugin may differ from the one you are encountering. Have you tried setting up the plugin in a fresh WP installation with the default theme, just for testing? This will help identify whether the issue lies with the website or the account setup.

Thanks.
